### DISCUSSION TITLE
Why can't I get detailed c code debugging information with cflags`-g`

### DISCUSSION THREAD

**Discussion by byxzone** [cilium/ebpf#838] [Q&A] [answered]
when i try to run my program,i get this error message:
`2022/11/04 05:09:06 loading objects: field XdpProg: program xdp_prog: load program: permission denied: invalid indirect read from stack R2 off -18+2 size 4 (99 line(s) omitted)`
but i can not locate this error in my code
in BCC,i can get detail error message , and it told me which line of c code is error

**Comment by byxzone** [comment]
`-g` is add to `$BPF_CFLAGS` but it dosen't work

**Comment by lmb** [comment, marked as answer]
Try following the example from the package documentation: https://pkg.go.dev/github.com/cilium/ebpf#example-Program-VerifierError Please note that you currently can't avoid the call to `errors.As` by directly formatting the error with `%+v`.

**Reply by ti-mo** [reply]
@byxzone Note that clang's `-g` flag controls the compiler's BTF and DWARF output. What you're looking for here is the kernel's verifier log, which is not directly related. BCC both compiles and loads the program, so this may be a source of confusion.

BPF programs compiled with the `-g` flag will contain line_info records, which are loaded into the kernel along the program itself. These will make the verifier log contain C statements along with the program's instructions, like so:

```
	my_func:
		 ; bpf_map_lookup_elem()
		0: Call FnMapLookupElem
		 ; abc = 42
		1: LdImmDW dst: r0 imm: 42
```
